### DISCUSSION TITLE
pibook site

### DISCUSSION THREAD

**Discussion by mxidakis** [sdy60-2024/sdy60#6] [General]
Θα μπορούσαν οι διαχειριστές να κάνουν fork το pibook/site για να ξεκινήσουμε;

**Comment by Marigian** [comment]
Καλησπέρα ήθελα να το ρωτήσω και εγώ αυτό, ποιός θα αναλάβει το κομμάτι αυτό? Σαν μέλη πλεον του organization έχουν όλοι τη δυνατοτητα να δημιουργήσουν repository σωστα? Αν όχι ενημερώστε με προκειμένου να δουμε τι αλλαγές χρειάζονται ....

**Comment by mxidakis** [comment]
Μαρίνα εγώ φαίνομαι σαν outside collaborator και δεν βλέπω τρόπο να δημιουργήσω repo στον οργανισμό

**Reply by Marigian** [reply]
Εχω στείλει σε όλους invitation στον οργανισμό σαν member όπου μπορεί να δημιουργηθεί repo. Αν το δει κάποιος και είναι οκ, επιβεβαιωστε το μου

**Reply by mxidakis** [reply]
οκ

**Reply by Pagiatis-Spiros** [reply]
ok

**Comment by mxidakis** [comment]
Κάποιος από τους διαχειριστές θα πρέπει να κάνει deploy το site στο netlify ή στο vercel με τα credentials του οργανισμού sdy60-2024 στο github. Εκτός αν δεν έχω καταλάβει σωστά τί πρέπει να κάνουμε.

**Reply by Marigian** [reply]
Καλησπέρα, νομίζω έχουν όλοι πλεον ίδια πρόσβαση, επομένως αν βρει κάποιος τρόπο το προχωράει λογικά δεν υπάρχει κάποιο limitation.

**Reply by apapost** [reply]
Μιχάλη νομίζω ότι έκανα κάτι αντίστοιχο!

**Reply by mxidakis** [reply]
Αποστόλη έχεις τη διεύθυνση για το site στο Netlify;

**Comment by mxidakis** [comment]
Καλησπέρα κάποιος πήρε δικαιώματα owner και έσβησε όλη τη δουλειά μας.  Μπορείτε να εξηγήσετε τί έγινε, ώστε να το διορθώσουμε;

**Comment by Pagiatis-Spiros** [comment]
Παιδιά έγινε από δικό μου λάθος, σήμερα το πρωί.
Έχω στείλει email στον κ.Χωριανόπουλο και τον έχω ενημερώσει για την βλακεία που έκανα.

Λυπάμαι πάρα πολύ και φυσικά θα δεχθώ όποιες συνέπειες υπάρξουν ΑΠΟΚΛΕΙΣΤΙΚΑ εγώ.

 Από το πρωί παιδεύομαι να το επαναφέρω και πιστεύω πως κάτι έχω κάνει
https://pibook-test.netlify.app/

Μιχάλη αν θέλεις πάρε με ένα τηλέφωνο 6945471204, πρέπει μάλλον να κάνεις ξανά deploy το pibook-site στο netify σου.


υγ.
Πέστε στο καθηγητή πως υπεύθυνος για αυτό που έγινε είμαι εγώ.
________________________________
Από: mxidakis ***@***.***>
Στάλθηκε: Παρασκευή, 20 Δεκεμβρίου 2024 6:56 μμ
Προς: sdy60-2024/sdy60 ***@***.***>
Κοιν.: ΠΑΓΙΑΤΗΣ ΣΠΥΡΙΔΩΝ ***@***.***>; Comment ***@***.***>
Θέμα: Re: [sdy60-2024/sdy60] pibook site (Discussion #6)


Καλησπέρα κάποιος πήρε δικαιώματα owner και έσβησε όλη τη δουλειά μας. Μπορείτε να εξηγήσετε τί έγινε, ώστε να το διορθώσουμε;

**Reply by mxidakis** [reply]
Σπύρο μπορείς σε παρακαλώ να μου πεις τί έκανες μπας και το διορθώσουμε;
Δεν χρειάζεται να τιμωρηθεί κάποιος, οι συνέπειες μας βαρύνουν όλους και λίγο περισσότερο τους διαχειριστές αλλά δεν είναι εκεί το θέμα. Είναι μια ομαδική δουλειά που πρέπει να φέρουμε εις πέρας.
Το να κάνεις deploy στο netlify το δικό σου αντίγραφο του repo από το βιβλίο δεν νομίζω οτι έχει κανένα νόημα, πέρα από το να πειραματιστείς με τις αλλαγές που θέλεις να κάνεις εσύ.

**Reply by Pagiatis-Spiros** [reply]
Δεν είναι το δικό μου αντίγραφο Μιχάλη, είναι το βιβλίο στον οργανισμό sdy60-2024.

Όποτε μπορέσεις πάρε με τηλ. 6945471204.

**Reply by mxidakis** [reply]
Σπύρο αν μας έλεγες και τί έκανες θα ήταν πιο εύκολο να το διορθώσουμε.

**Reply by Pagiatis-Spiros** [reply]
Μιχάλη πραγματικά δεν μπορώ να καταλάβω τι είναι αυτό που εννοείς.

Σου είχα αναφέρει τόσες φορές, ότi τον μοναδικό που χρειάζονταν (αρχικά) ήταν να γίνει cache clear and deploy again στο netlify.
Η μόνη διαφορά αυτού που ανέβασες με αυτό που σου είπα εγώ να κάνεις deploy, είναι η αρχική σελίδα με τις 10 γραμμές των οδηγιών σου.

Παρόλα αυτά προσπάθησα πάμπολες φορές να εποκοινωνήσω μαζί σου για να βρούμε μια λύση η οποία δεν θα επειρέαζε την ομάδα,
αλλά πάντα έβρισκα τοίχο (δεν γνωρίζω το  λόγο).

Σε πείρα τηλέφωνο στο κινητό σου (δεν απαντούσες δεν με ξαναπείρες).
Σε πείρα τηλέφωνο στο viber (τα ίδια).
Έστειλα μήνυμα για να επικοινωνήσουμε στο study (τα ίδια).
Έστειλα μήνυμα στο github για επικοινωνία (τα ίδια). 

Δεν είχα ιδέα τι άλλο να κάνω, το ξέρω πως είναι πολύ σπαστικό κάποιος να χαλάει την δουλειά σου αλλά το θέμα είναι να βρεθεί μια λύση για την ομάδα.

**Reply by mxidakis** [reply]
Σπύρο δεν καταλαβαίνω τον εκνευρισμό σου και δεν μπορώ να καταλάβω τί δεν καταλαβαίνεις. Σου έχω ζητήσει τρείς φορές να μας πεις τί έκανες και χάθηκε η δουλειά των συμφοιτητών μας στο site και το μόνο που λες είναι οτι έκανες λάθος.  Κατά τα άλλα ευχαριστώ για τις υποδείξεις και ευχαριστώ για την αναγνώριση οτι έγραψα και 10 γραμμές.
ΥΓ. Δεν έχω πρόσβαση στο netlify.

**Comment by mxidakis** [comment]
The site is back on

**Reply by mxidakis** [reply]
Please check your work and resubmit it, if something is missing.

**Reply by Pagiatis-Spiros** [reply]
Τα δικά μου αρχεία είναι ok!

**Comment by mxidakis** [comment]
11:22:35 PM: Waiting for other deploys from your team to complete. Check the queue: https://app.netlify.com/teams/vasiliskokkalis/builds
11:23:34 PM: Build ready to start
11:23:53 PM: build-image version: 9c9fb6952e50bb092d4b66daf2368677e5c68e34 (focal)
11:23:53 PM: buildbot version: 9c9fb6952e50bb092d4b66daf2368677e5c68e34
11:23:53 PM: Fetching cached dependencies
11:23:53 PM: Starting to download cache of 366.9MB
11:23:56 PM: Finished downloading cache in 2.965s
11:23:56 PM: Starting to extract cache
11:24:02 PM: Finished extracting cache in 5.636s
11:24:02 PM: Finished fetching cache in 8.666s
11:24:02 PM: Starting to prepare the repo for build
11:24:03 PM: Preparing Git Reference pull/93/head
11:24:06 PM: Starting to install dependencies
11:24:06 PM: Started restoring cached mise cache
11:24:07 PM: Finished restoring cached mise cache
11:24:09 PM: mise python@3.13.2   install
11:24:09 PM: mise python@3.13.2   download cpython-3.13.2+20250212-x86_64-unknown-linux-gnu-install_only_stripped.tar.gz
11:24:10 PM: mise python@3.13.2   extract cpython-3.13.2+20250212-x86_64-unknown-linux-gnu-install_only_stripped.tar.gz
11:24:10 PM: mise python@3.13.2   python --version
11:24:10 PM: mise python@3.13.2   Python 3.13.2
11:24:10 PM: mise python@3.13.2 ✓ installed
11:24:10 PM: Python version set to 3.13
11:24:12 PM: Collecting pipenv
11:24:12 PM:   Downloading pipenv-2024.4.1-py3-none-any.whl.metadata (17 kB)
11:24:12 PM: Collecting certifi (from pipenv)
11:24:12 PM:   Downloading certifi-2025.1.31-py3-none-any.whl.metadata (2.5 kB)
11:24:12 PM: Collecting packaging>=22 (from pipenv)
11:24:12 PM:   Downloading packaging-24.2-py3-none-any.whl.metadata (3.2 kB)
11:24:13 PM: Collecting setuptools>=67 (from pipenv)
11:24:13 PM:   Downloading setuptools-75.8.0-py3-none-any.whl.metadata (6.7 kB)
11:24:13 PM: Collecting virtualenv>=20.24.2 (from pipenv)
11:24:13 PM:   Downloading virtualenv-20.29.2-py3-none-any.whl.metadata (4.5 kB)
11:24:13 PM: Collecting distlib<1,>=0.3.7 (from virtualenv>=20.24.2->pipenv)
11:24:13 PM:   Downloading distlib-0.3.9-py2.py3-none-any.whl.metadata (5.2 kB)
11:24:13 PM: Collecting filelock<4,>=3.12.2 (from virtualenv>=20.24.2->pipenv)
11:24:13 PM:   Downloading filelock-3.17.0-py3-none-any.whl.metadata (2.9 kB)
11:24:13 PM: Collecting platformdirs<5,>=3.9.1 (from virtualenv>=20.24.2->pipenv)
11:24:13 PM:   Downloading platformdirs-4.3.6-py3-none-any.whl.metadata (11 kB)
11:24:13 PM: Downloading pipenv-2024.4.1-py3-none-any.whl (3.0 MB)
11:24:13 PM:    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.0/3.0 MB 60.6 MB/s eta 0:00:00
11:24:13 PM: Downloading packaging-24.2-py3-none-any.whl (65 kB)
11:24:13 PM: Downloading setuptools-75.8.0-py3-none-any.whl (1.2 MB)
11:24:13 PM:    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.2/1.2 MB 75.3 MB/s eta 0:00:00
11:24:13 PM: Downloading virtualenv-20.29.2-py3-none-any.whl (4.3 MB)
11:24:13 PM:    ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.3/4.3 MB 118.5 MB/s eta 0:00:00
11:24:13 PM: Downloading certifi-2025.1.31-py3-none-any.whl (166 kB)
11:24:13 PM: Downloading distlib-0.3.9-py2.py3-none-any.whl (468 kB)
11:24:13 PM: Downloading filelock-3.17.0-py3-none-any.whl (16 kB)
11:24:13 PM: Downloading platformdirs-4.3.6-py3-none-any.whl (18 kB)
11:24:13 PM: Installing collected packages: distlib, setuptools, platformdirs, packaging, filelock, certifi, virtualenv, pipenv
11:24:16 PM: Successfully installed certifi-2025.1.31 distlib-0.3.9 filelock-3.17.0 packaging-24.2 pipenv-2024.4.1 platformdirs-4.3.6 setuptools-75.8.0 virtualenv-20.29.2
11:24:16 PM: [notice] A new release of pip is available: 24.3.1 -> 25.0.1
11:24:16 PM: [notice] To update, run: pip install --upgrade pip
11:24:16 PM: Required ruby-3.1.0 is not installed.
11:24:16 PM: To install do: 'rvm install "ruby-3.1.0"'
11:24:16 PM: Attempting Ruby version 3.1.0, read from .ruby-version file
11:24:17 PM: Required ruby-3.1.0 is not installed - installing.
11:24:17 PM: Searching for binary rubies, this might take some time.
11:24:17 PM: Found remote file https://rvm_io.global.ssl.fastly.net/binaries/ubuntu/20.04/x86_64/ruby-3.1.0.tar.bz2
11:24:17 PM: Checking requirements for ubuntu.
11:24:18 PM: Requirements installation successful.
11:24:18 PM: ruby-3.1.0 - #configure
11:24:18 PM: ruby-3.1.0 - #download
11:24:18 PM: No checksum for downloaded archive, recording checksum in user configuration.
11:24:18 PM: ruby-3.1.0 - #validate archive
11:24:23 PM: ruby-3.1.0 - #extract
11:24:27 PM: ruby-3.1.0 - #validate binary
11:24:27 PM: ruby-3.1.0 - #setup
11:24:28 PM: ruby-3.1.0 - #gemset created /opt/buildhome/.rvm/gems/ruby-3.1.0@global
11:24:28 PM: ruby-3.1.0 - #importing gemset /opt/buildhome/.rvm/gemsets/global.gems........................................
11:24:28 PM: ruby-3.1.0 - #generating global wrappers........
11:24:28 PM: ruby-3.1.0 - #gemset created /opt/buildhome/.rvm/gems/ruby-3.1.0
11:24:28 PM: ruby-3.1.0 - #importing gemsetfile /opt/buildhome/.rvm/gemsets/default.gems evaluated to empty gem list
11:24:29 PM: ruby-3.1.0 - #generating default wrappers........
11:24:29 PM: Using /opt/buildhome/.rvm/gems/ruby-3.1.0
11:24:29 PM: Using Ruby version 3.1.0
11:24:30 PM: Started restoring cached ruby gems
11:24:30 PM: Finished restoring cached ruby gems
11:24:30 PM: Installing gem bundle
11:24:30 PM: [DEPRECATED] The `--path` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set --local path '/opt/build/cache/bundle'`, and stop using this flag
11:24:30 PM: [DEPRECATED] The --binstubs option will be removed in favor of `bundle binstubs --all`
11:24:32 PM: Fetching gem metadata from https://rubygems.org/..........
11:24:32 PM: Resolving dependencies....
11:24:32 PM: Using rake 13.2.1
11:24:32 PM: Using connection_pool 2.5.0
11:24:32 PM: Using securerandom 0.4.1
11:24:32 PM: Using public_suffix 6.0.1
11:24:32 PM: Using json 2.10.1
11:24:32 PM: Using mini_portile2 2.8.8
11:24:32 PM: Fetching mutex_m 0.3.0
11:24:32 PM: Using bundler 2.3.3
11:24:32 PM: Using forwardable 1.3.3
11:24:32 PM: Using observer 0.1.2
11:24:32 PM: Fetching rexml 3.4.1
11:24:32 PM: Using http_parser.rb 0.8.0
11:24:32 PM: Using forwardable-extended 2.6.0
11:24:32 PM: Using rouge 4.5.1
11:24:32 PM: Using base64 0.2.0
11:24:32 PM: Using benchmark 0.4.0
11:24:32 PM: Using bigdecimal 3.1.9
11:24:32 PM: Using concurrent-ruby 1.3.5
11:24:32 PM: Using drb 2.2.1
11:24:32 PM: Fetching logger 1.6.6
11:24:32 PM: Using minitest 5.25.4
11:24:32 PM: Using racc 1.8.1
11:24:32 PM: Using latex-decode 0.4.0
11:24:32 PM: Using date 3.4.1
11:24:32 PM: Using uri 1.0.2
11:24:32 PM: Using colorator 1.1.0
11:24:32 PM: Using csv 3.3.2
11:24:32 PM: Using eventmachine 1.2.7
11:24:32 PM: Using ffi 1.17.1
11:24:32 PM: Using filesize 0.2.0
11:24:32 PM: Using gemoji 4.1.0
11:24:32 PM: Using rb-fsevent 0.11.2
11:24:32 PM: Using liquid 4.0.4
11:24:32 PM: Using mercenary 0.4.0
11:24:32 PM: Fetching stringio 3.1.5
11:24:32 PM: Using safe_yaml 1.0.5
11:24:32 PM: Using unicode-display_width 2.6.0
11:24:32 PM: Using webrick 1.9.1
11:24:32 PM: Using progressbar 1.13.0
11:24:32 PM: Using verbal_expressions 0.1.5
11:24:32 PM: Using jekyll-paginate 1.1.0
11:24:32 PM: Using rubyzip 2.4.1
11:24:32 PM: Using addressable 2.8.7
11:24:32 PM: Using google-protobuf 4.29.3 (x86_64-linux)
11:24:32 PM: Using i18n 1.14.7
11:24:32 PM: Using tzinfo 2.0.6
11:24:32 PM: Using pathutil 0.16.2
11:24:32 PM: Fetching sass-embedded 1.85.0
11:24:32 PM: Using tzinfo-data 1.2025.1
11:24:32 PM: Using net-http 0.6.0
11:24:32 PM: Using em-websocket 0.5.3
11:24:33 PM: Using faraday-net_http 3.4.0
11:24:33 PM: Using rb-inotify 0.11.1
11:24:33 PM: Using terminal-table 3.0.2
11:24:33 PM: Using time 0.4.1
11:24:33 PM: Using listen 3.9.0
11:24:33 PM: Fetching nokogiri 1.18.3
11:24:33 PM: Installing mutex_m 0.3.0
11:24:33 PM: Using namae 1.2.0
11:24:33 PM: Using bibtex-ruby 6.1.0
11:24:33 PM: Using jekyll-watch 2.2.1
11:24:33 PM: Installing stringio 3.1.5 with native extensions
11:24:33 PM: Installing logger 1.6.6
11:24:33 PM: Installing sass-embedded 1.85.0 with native extensions
11:24:33 PM: Installing rexml 3.4.1
11:24:33 PM: Fetching httpclient 2.9.0
11:24:33 PM: Installing nokogiri 1.18.3 with native extensions
11:24:33 PM: Using activesupport 7.2.2.1
11:24:33 PM: Using faraday 2.12.2
11:24:33 PM: Using sawyer 0.9.2
11:24:33 PM: Using octokit 4.25.1
11:24:33 PM: Using jekyll-gist 1.5.0
11:24:33 PM: Installing httpclient 2.9.0
11:24:33 PM: Using csl 1.6.0
11:24:33 PM: Using kramdown 2.5.1
11:24:33 PM: Using csl-styles 1.0.1.11
11:24:33 PM: Using kramdown-parser-gfm 1.1.0
11:24:33 PM: Using algoliasearch 1.27.5
11:24:34 PM: Using open-uri 0.5.0
11:24:34 PM: Using citeproc 1.1.0
11:24:34 PM: Using citeproc-ruby 1.1.14
11:25:33 PM: Failed during stage 'Install dependencies': dependency_installation script returned non-zero exit code: 1
11:25:33 PM: Using algolia_html_extractor 2.6.4
11:25:33 PM: Using html-pipeline 2.14.3
11:25:33 PM: Gem::Ext::BuildError: ERROR: Failed to build gem native extension.
11:25:33 PM: current directory:
11:25:33 PM: /opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass
11:25:33 PM: /opt/buildhome/.rvm/rubies/ruby-3.1.0/bin/ruby
11:25:33 PM: -I/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0 -rrubygems
11:25:33 PM: /opt/build/cache/bundle/ruby/3.1.0/gems/rake-13.2.1/exe/rake
11:25:33 PM: RUBYARCHDIR\=/opt/build/cache/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/sass-embedded-1.85.0
11:25:33 PM: RUBYLIBDIR\=/opt/build/cache/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/sass-embedded-1.85.0
11:25:33 PM: gem install --force --install-dir
11:25:33 PM: /opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/ruby
11:25:33 PM: --no-document --ignore-dependencies --platform x86_64-linux-gnu --version 1.85.0
11:25:33 PM: sass-embedded
11:25:33 PM: rake aborted!
11:25:33 PM: NameError: uninitialized constant JSON::Fragment
11:25:33 PM: <internal:/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in
11:25:33 PM: `require'
<internal:/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in
`require'
<internal:/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in
11:25:33 PM: `require'
<internal:/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in
`require'
<internal:/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in
11:25:33 PM: `require'
<internal:/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in
`require'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:567:in
11:25:33 PM: `package_json'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:573:in
`dart_sass_version'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:579:in
11:25:33 PM: `dart_sass'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:52:in
`rescue in block in <top (required)>'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:48:in
11:25:33 PM: `block in <top (required)>'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:72:in
`block in <top (required)>'
/opt/build/cache/bundle/ruby/3.1.0/gems/rake-13.2.1/exe/rake:27:in `<main>'

Caused by:
Gem::UnsatisfiableDependencyError: Unable to resolve dependency: user requested
'sass-embedded (= 1.85.0)'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:413:in
`gem_install'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:48:in
11:25:33 PM: `block in <top (required)>'
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/Rakefile:72:in
`block in <top (required)>'
/opt/build/cache/bundle/ruby/3.1.0/gems/rake-13.2.1/exe/rake:27:in `<main>'
Tasks: TOP => dart-sass/sass
(See full trace by running task with --trace)
rm -rf
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0/ext/sass/ruby

rake failed, exit code 1

Gem files will remain installed in
/opt/build/cache/bundle/ruby/3.1.0/gems/sass-embedded-1.85.0 for inspection.
Results logged to
/opt/build/cache/bundle/ruby/3.1.0/extensions/x86_64-linux/3.1.0/sass-embedded-1.85.0/gem_make.out

/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/ext/builder.rb:95:in
`run'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/ext/rake_builder.rb:28:in
11:25:33 PM: `build'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/ext/builder.rb:161:in
`build_extension'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/ext/builder.rb:195:in
11:25:33 PM: `block in build_extensions'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in
`each'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/ext/builder.rb:192:in
11:25:33 PM: `build_extensions'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/rubygems/installer.rb:847:in
`build_extensions'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/rubygems_gem_installer.rb:71:in
11:25:33 PM: `build_extensions'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/rubygems_gem_installer.rb:28:in
`install'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/source/rubygems.rb:204:in
11:25:33 PM: `install'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/installer/gem_installer.rb:54:in
`install'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/installer/gem_installer.rb:16:in
11:25:33 PM: `install_from_spec'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/installer/parallel_installer.rb:186:in
`do_install'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/installer/parallel_installer.rb:177:in
11:25:33 PM: `block in worker_pool'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/worker.rb:62:in
`apply_func'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/worker.rb:57:in
11:25:33 PM: `block in process_queue'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/worker.rb:54:in
`loop'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/worker.rb:54:in
11:25:33 PM: `process_queue'
/opt/buildhome/.rvm/rubies/ruby-3.1.0/lib/ruby/3.1.0/bundler/worker.rb:91:in
`block (2 levels) in create_threads'
11:25:33 PM: An error occurred while installing sass-embedded (1.85.0), and Bundler cannot
11:25:33 PM: continue.
11:25:33 PM: In Gemfile:
11:25:33 PM:   jekyll-algolia was resolved to 1.7.1, which depends on
11:25:33 PM:     jekyll was resolved to 4.4.1, which depends on
11:25:33 PM:       jekyll-sass-converter was resolved to 3.0.0, which depends on
11:25:33 PM:         sass-embedded
11:25:33 PM: Error during gem install
11:25:33 PM: Failing build: Failed to install dependencies
11:25:33 PM: Finished processing build request in 1m39.658s

**Reply by mxidakis** [reply]
Φαίνεται να υπάρχει πρόβλημα με το sass-embedded (1.85.00) gem. Μάλλον έχει γίνει update αλλά δεν δουλεύει. Το θέμα είναι γιατί αφού δεν δουλεύει έκανε update το Gemfile.lock. Θα δουλέψει αν βάλουμε ένα παλιότερο αρχείο Gemfile.lock στο repo ή να τρέξουμε ένα bundle install τοπικά;

**Comment by Pagiatis-Spiros** [comment]
Το Jekyll είναι φτιαγμένο με Ruby, το sass-embedded είναι ένα ενσωματωμένο plugin αυτού ([δες εδώ](https://jekyllrb.com/docs/configuration/sass/)).
Το θέμα που δημιουργήθηκε πιθανότατα έχει να κάνει με κάποια ενημερωμένη έκδοση κάποιου dependency, εμείς χρησιμοποιούμε τα παρακάτω (από το gemfile)

```
group :jekyll_plugins do
  gem "jekyll-paginate"
  gem "jekyll-sitemap"
  gem "jekyll-gist"
  gem "jekyll-feed"
  gem "jemoji"
  gem "jekyll-include-cache"
  gem "jekyll-algolia"
  gem 'jekyll-tagories'
  gem 'jekyll-remote-theme'
  gem 'jekyll-scholar'
end
```


Οπότε αν θέλεις να δεις ποιο φταίει, πρέπει να απενεργοποιείς ένα κάθε φορά και να κάνεις clear cache and deploy again στο netlify.
Μετά εφόσον βρεις το ελαττωματικό, μπορείς να κάνεις local install μια ποιο παλιά version και λογικά θα δουλέψει κανονικά (πολύ χρονοβόρα λύση),


Έναν άλλο τρόπο που βρήκα είναι να αναβαθμίσουμε την Ruby, αυτήν την στιγμή αν δεις  στο log που ανέβασες τρέχει την 3.1.0 που είναι πάρα πολύ παλιά 25-12-2021 ([Ruby releases](https://www.ruby-lang.org/en/downloads/releases/)).
Συνεπώς το πρώτο που πρέπει να γίνει είναι να τρέξουμε μια ποιο νέα version, αλλά επειδή δεν γνωρίζω αν θα υπάρξει κάποιο πρόβλημα συμβατότητας με τις τελευταίες καλό είναι να πάμε σε μια παλιά (αλλά ποιο νέα από την εγκαταστημένη).
Εγώ έκανα δοκιμή με την 3.2.0 και δούλεψε κανονικά, παρακάτω γράφω τα βήματα για να γίνει αυτό.

- Πάμε στο root του pibook και βρίσκουμε τον φάκελο .ruby-version 
- Όταν τον ανοίξουμε θα δούμε την version που σε κάθε deploy κατεβάζει και κάνει install το netlify (στην περίπτωση μας αναφέρεται η 3.1.0)
- Κάνουμε edit και το αλλάζουμε σε 3.2.0 
- Πάμε στο netlify και κάνουμε clear cache and deploy again το pibook

υγ.
Αν δεν υπάρχει χρόνος να κάνετε τα παραπάνω, δώστε μου τα credentials από το nεtlify να το κάνω εγώ.
